Algorithm.Framework/Alphas/MacdAlphaModel.cs

```diff
@@ -66,16 +75,17 @@ public IEnumerable<Insight> Update(QCAlgorithmFramework algorithm, Slice data)
 
                 var direction = InsightDirection.Flat;
                 var normalizedSignal = sd.MACD.Signal / sd.Security.Price;
-                if (normalizedSignal > _bounceThresholdPercent)
+                if (normalizedSignal > BounceThresholdPercent)
                 {
                     direction = InsightDirection.Up;
                 }
-                else if (normalizedSignal < -_bounceThresholdPercent)
+                else if (normalizedSignal < -BounceThresholdPercent)
                 {
                     direction = InsightDirection.Down;
                 }
 
-                var insight = new Insight(sd.Security.Symbol, InsightType.Price, direction, _insightPeriod);
+                var insightPeriod = sd.DataResolution.Multiply(_fastPeriod);
+                var insight = new Insight(sd.Security.Symbol, InsightType.Price, direction, insightPeriod);
                 if (insight.Equals(sd.PreviousInsight))
```

Comment:
Note for future - have to make sure generated time isn't compared here. Or include generated time and change this comparison method.

Reply:
fyi, opened issue to address this: https://github.com/QuantConnect/Lean/issues/1829